src/compiler/parser.ts

```diff
@@ -1157,16 +1163,7 @@ namespace ts {
                 parseErrorAtCurrentToken(diagnosticMessage, arg0);
             }
 
-            const result = createNode(kind, scanner.getStartPos());
-            switch (kind) {
-                case SyntaxKind.Identifier:
-                    (result as Identifier).escapedText = "" as __String;
-                    break;
-                default: // TODO: GH#17346

```

Comment:
To preserve the intent of current behaviors, why not just replace the switch with
```
if (kind === SyntaxKind.Identifier) {
    (result as Identifier).escapedText = "" as __String;
}
else if (isLiteralKind(kind)) {
    (result as LiteralLikeNode).text = "";
}
```
?
This would also preserve the behavior when `createMissingNode` is called via `parseExpected`, or other indirect calls. We could also add some overloads for specific syntax kinds to avoid the need for casts.